include/boost/geometry/extensions/gis/io/wkb/detail/writer.hpp

```diff
@@ -0,0 +1,217 @@
+		struct value_writer
+		{
+			typedef T value_type;
+
+			template <typename OutputIterator>
+			static bool write(const T& value, OutputIterator& iter, byte_order_type::enum_t byte_order)
+			{
+				endian::endian_value<T> parsed_value(value);
+
+// 				if (byte_order_type::xdr == byte_order)
+// 				{
+// 					parsed_value.template store<endian::big_endian_tag>(iter);
+// 				}
+// 				else if (byte_order_type::ndr == byte_order)
+// 				{
+// 					parsed_value.template store<endian::little_endian_tag>(iter);
+// 				}
+// 				else
+// 				{
+					parsed_value.template store<endian::native_endian_tag>(iter);
+// 				}
+
+				return true;
+			}
+		};
+
+		template <typename P, int I, int N>
+		struct writer_assigner
+		{
+			template <typename OutputIterator>
+			static void run(const P& point, OutputIterator& iter, byte_order_type::enum_t byte_order)
+			{
+				typedef typename coordinate_type<P>::type coordinate_type;
+
+				value_writer<double>::write(boost::geometry::get<I>(point), iter, byte_order);
+
+				writer_assigner<P, I+1, N>::run(point, iter, byte_order);
+			}
+		};
+
+		template <typename P, int N>
+		struct writer_assigner<P, N, N>
+		{
+			template <typename OutputIterator>
+			static void run(const P& point, OutputIterator& iter, byte_order_type::enum_t byte_order)
+			{
+				// terminate
+				boost::ignore_unused_variable_warning(point);
+				boost::ignore_unused_variable_warning(iter);
+				boost::ignore_unused_variable_warning(byte_order);
+			}
+		};
+
+		template <typename P>
+		struct point_writer
+		{
+			template <typename OutputIterator>
+			static bool write(const P& point, OutputIterator& iter,
+				byte_order_type::enum_t byte_order)
+			{
+				// write endian type
+
+				value_writer<uint8_t>::write(byte_order, iter, byte_order);
+
+				// write geometry type
+				uint32_t type = geometry_type<P>::get();
+				value_writer<uint32_t>::write(type, iter, byte_order);
+
+				// write point's x, y, z
+
+				writer_assigner<P, 0, dimension<P>::value>::run(point, iter, byte_order);
+
+				return true;
+			}
+		};
+
+		template <typename L>
+		struct linestring_writer
+		{
+			template <typename OutputIterator>
+			static bool write(const L& linestring, OutputIterator& iter,
+				byte_order_type::enum_t byte_order)
+			{
+				// write endian type
+
+				value_writer<uint8_t>::write(byte_order, iter, byte_order);
+
+				// write geometry type
+				uint32_t type = geometry_type<L>::get();
+				value_writer<uint32_t>::write(type, iter, byte_order);
+
+				// write num points
+
+				uint32_t num_points = boost::size(linestring);
+				value_writer<uint32_t>::write(num_points, iter, byte_order);
+
+				typedef typename point_type<L>::type point_type;
+
+				for(boost::range_iterator<const L>::type point_iter = boost::begin(linestring); point_iter != boost::end(linestring); ++point_iter)
+				{
+					// write point's x, y, z
+					writer_assigner<point_type, 0, dimension<point_type>::value>::run(*point_iter, iter, byte_order);
```

Comment:
This and 133 lines are probably too long. In such cases we put the for-loop expressions or template parameters in separate lines, e.g.:

    for(boost::range_iterator<const L>::type point_iter = boost::begin(linestring);
        point_iter != boost::end(linestring);
        ++point_iter)
    {
        writer_assigner
            <
                point_type, 0, dimension<point_type>::value
            >::run(*point_iter, iter, byte_order);
    }